tls/s2n_x509_validator.c

```diff
@@ -371,6 +371,39 @@ s2n_cert_validation_code s2n_x509_validator_validate_cert_chain(struct s2n_x509_
 
         S2N_ERROR_IF(op_code <= 0, S2N_ERR_CERT_UNTRUSTED);
         validator->state = VALIDATED;
+
+        /* Check to ensure certificate signatures appear in certificate_signature_preferences.
+         * Additionally we only want this check to apply in TLS1.3 */
+        if (conn->config->security_policy->certificate_signature_preferences &&
+                conn->actual_protocol_version >= S2N_TLS13) {
+            
+            STACK_OF(X509) *validated_chain = X509_STORE_CTX_get1_chain(validator->store_ctx);
+
+            if (!validated_chain) {
+                goto clean_up;
+            }
+
+            const int certs_in_chain = sk_X509_num(validated_chain);
+
+            if (!certs_in_chain) {
+                goto clean_up;
+            }
+
+            /* Do not validate the root certificate */
+            for (int i = 0; i < certs_in_chain - 1; i++) {
```

Comment:
Usually we'd say "use size_t" here, but since sk_x509_num returns an int instead of size_t, I think you might be right to use "int" instead.

Also, the comment here might be clearer if you explicitly do "int certs_to_validate = certs_in_chain - 1" before the loop and put the comment on that line instead.